modules/cli/src/main/scala/scala/cli/exportCmd/MavenProjectDescriptor.scala

```diff
@@ -0,0 +1,316 @@
+package scala.cli.exportCmd
+
+import coursier.ivy.IvyRepository
+import coursier.maven.MavenRepository
+import coursier.parse.RepositoryParser
+import dependency.{AnyDependency, NoAttributes, ScalaNameAttributes}
+
+import java.nio.charset.StandardCharsets
+import java.nio.file.Path
+
+import scala.build.errors.BuildException
+import scala.build.internal.Constants
+import scala.build.internal.Runner.frameworkName
+import scala.build.options.{BuildOptions, Platform, Scope, ShadowingSeq}
+import scala.build.testrunner.AsmTestRunner
+import scala.build.{Logger, Positioned, Sources}
+import scala.cli.ScalaCli
+import scala.cli.exportCmd.POMBuilderHelper.*
+import scala.xml.{Elem, XML}
+
+object POMBuilderHelper {
+  def buildNode(name: String, value: String): Elem =
+    new Elem(
+      null,
+      name,
+      scala.xml.Null,
+      scala.xml.TopScope,
+      minimizeEmpty = false,
+      scala.xml.Text(value)
+    )
+}
+
+final case class MavenProjectDescriptor(
+  mavenPluginVersion: String,
+  mavenScalaPluginVersion: String,
+  mavenExecPluginVersion: String,
+  extraSettings: Seq[String],
+  logger: Logger
+) extends ProjectDescriptor {
+  private val q  = "\""
+  private val nl = System.lineSeparator()
+
+  private def sources(sourcesMain: Sources, sourcesTest: Sources): MavenProject = {
+    val mainSources = ProjectDescriptor.sources(sourcesMain)
+    val testSources = ProjectDescriptor.sources(sourcesTest)
+    MavenProject(
+      mainSources = mainSources,
+      testSources = testSources
+    )
+  }
+
+  // todo: fill this
+  private def javaOptionsSettings(options: BuildOptions): MavenProject =
+    MavenProject(
+      settings = Nil
+    )
+
+  private def javacOptionsSettings(options: BuildOptions): List[String] = {
+
+    val javacOptionsSettings =
+      if (options.javaOptions.javacOptions.toSeq.isEmpty) Nil
+      else {
+        val options0 = options
+          .javaOptions
+          .javacOptions
+          .toSeq
+          .map(_.value)
+          .map(o => "\"" + o.replace("\"", "\\\"") + "\"")
+        options0
+      }
+
+    javacOptionsSettings.toList
+  }
+
+  private def projectArtifactSettings(): MavenProject =
+    // todo: Is it really needed to configure the values, or always use the default?
+    MavenProject(
+      groupId = None,
+      artifactId = None,
+      version = None
+    )
+
+  private def dependencySettings(
+    options: BuildOptions,
+    testOptions: BuildOptions,
+    scope: Scope,
+    sources: Sources
+  ): MavenProject = {
+
+    val scalaV = getScalaVersion(options)
+    def getScalaMajorPrefix =
```

Comment:
```suggestion
    def getScalaPrefix =
```
`major` implies the major version component, which would be `2` for `2.12.x`